### PR TITLE
Dynamic reloading after voting

### PR DESCRIPTION
**Pull request by carlosgj94**

### CODE REVIEW

File: plugins/dualGovernance/hooks/useProposal.tsx [hunk 1]
```diff
@@ -36,17 +37,19 @@ export function useProposal(
...
+  const { data: blockNumber } = useBlockNumber({ watch: true });
 
   // Proposal on-chain data
   const {
     data: proposalResult,
     error: proposalError,
     fetchStatus: proposalFetchStatus,
-    refetch: proposalRefetch
+    refetch: proposalRefetch,
+    queryKey: proposalQueryKey,
```
Comment (brickpop) [inline plugins/dualGovernance/hooks/useProposal.tsx:52]: Where is this variable declared?

File: plugins/dualGovernance/hooks/useUserCanVeto.tsx [hunk 2]
```diff
@@ -16,8 +17,17 @@ export function useUserCanVeto(proposalId: bigint) {
...
+    if (Number(blockNumber?.toString()) % 2 === 0) {
+      canVetoRefetch();
+    }
+  }, [blockNumber])
+
+  useEffect(() => {
+    if (forceRefetch && !userVetoed) {
+      canVetoRefetch();
+    }
+    setUserVetoed(true)
```
Comment (brickpop) [inline plugins/dualGovernance/hooks/useUserCanVeto.tsx:29]: Why setting it to true unconditionally?

File: plugins/dualGovernance/hooks/useUserCanVeto.tsx [hunk 3]
```diff
@@ -16,8 +17,17 @@ export function useUserCanVeto(proposalId: bigint) {
   });
 
   useEffect(() => {
-    canVetoRefetch();
-  }, [blockNumber]);
+    if (Number(blockNumber?.toString()) % 2 === 0) {
+      canVetoRefetch();
+    }
+  }, [blockNumber])
```
Comment (brickpop) [inline plugins/dualGovernance/hooks/useUserCanVeto.tsx:23]:
```suggestion
    canVetoRefetch();
  }, [Number(blockNumber) % 2 === 0)
```

File: plugins/dualGovernance/hooks/useProposal.tsx [hunk 4]
```diff
@@ -12,6 +12,7 @@ import {
 } from "@/plugins/dualGovernance/utils/types";
 import { useQuery } from "@tanstack/react-query";
 import { PUB_CHAIN } from "@/constants";
+import { useQueryClient } from '@tanstack/react-query'
```
Comment (brickpop) [inline plugins/dualGovernance/hooks/useProposal.tsx:15]:
```suggestion
```

File: plugins/dualGovernance/hooks/useProposal.tsx [hunk 5]
```diff
@@ -36,17 +37,19 @@ export function useProposal(
   proposalId: string,
   autoRefresh = false
 ) {
+  const queryClient = useQueryClient()
```
[the remark this answers is not in the document]
Comment (brickpop) [reply plugins/dualGovernance/hooks/useProposal.tsx:40]:
```suggestion
```

File: plugins/dualGovernance/hooks/useProposal.tsx (hunk 1 above)
Comment (brickpop) [reply plugins/dualGovernance/hooks/useProposal.tsx:52]:
```suggestion
```

File: plugins/dualGovernance/hooks/useProposalVeto.tsx [hunk 6]
```diff
@@ -13,7 +13,6 @@ import { useAlertContext, AlertContextProps } from "@/context/AlertContext";
 import { PUB_CHAIN, PUB_DUAL_GOVERNANCE_PLUGIN_ADDRESS } from "@/constants";
 
 export function useProposalVeto(proposalId: string) {
-  const { reload } = useRouter();
```
Comment (brickpop) [inline plugins/dualGovernance/hooks/useProposalVeto.tsx:16]: We should also remove the `useRouter` import above

File: plugins/dualGovernance/hooks/useUserCanVeto.tsx [hunk 7]
```diff
@@ -1,13 +1,14 @@
...
 import { PUB_CHAIN, PUB_DUAL_GOVERNANCE_PLUGIN_ADDRESS } from "@/constants";
 
-export function useUserCanVeto(proposalId: bigint) {
+export function useUserCanVeto(proposalId: bigint, forceRefetch: boolean) {
   const { address } = useAccount();
   const { data: blockNumber } = useBlockNumber({ watch: true });
+  const [userVetoed, setUserVetoed] = useState<boolean>(false)
 
-  const { data: canVeto, refetch: canVetoRefetch } = useReadContract({
+  const { data: canVeto, refetch: canVetoRefetch, queryKey } = useReadContract({
```
Comment (brickpop) [inline plugins/dualGovernance/hooks/useUserCanVeto.tsx:11]:
```suggestion
  const { data: canVeto, refetch: canVetoRefetch } = useReadContract({
```

File: plugins/dualGovernance/hooks/useUserCanVeto.tsx [hunk 8]
```diff
@@ -16,8 +17,17 @@ export function useUserCanVeto(proposalId: bigint) {
   });
 
   useEffect(() => {
-    canVetoRefetch();
-  }, [blockNumber]);
+    if (Number(blockNumber?.toString()) % 2 === 0) {
```
Comment (brickpop) [inline plugins/dualGovernance/hooks/useUserCanVeto.tsx:20]:
```suggestion
    if (Number(blockNumber) % 2 === 0) {
```

**Comment by brickpop** [comment]
While the current forceRefresh on userCanVeto seems to work I would strongly recommend to do the refetching at the `useProposal()` of the parent component, not at the child one. 

This is:
- `useUserCanVeto()` should simply retrieve the current proposal state and digest the output. No external artifacts
- `useProposalVeto()` could be calling `refetch()` on `useProposal`, so that everything else is properly repopulated and updated

File: plugins/dualGovernance/hooks/useUserCanVeto.tsx (hunk 2 above)
Comment (carlosgj94) [reply plugins/dualGovernance/hooks/useUserCanVeto.tsx]: Removed

**Pull request merged**